content/docs/reference-react-dom-server.md

```diff
@@ -40,47 +40,91 @@ The following methods can be used in the environments that don't support streams
 ReactDOMServer.renderToPipeableStream(element, options)
 ```
 
-Render a React element to its initial HTML. Returns a stream with a `pipe(res)` method to pipe the output and `abort()` to abort the request. Fully supports Suspense and streaming of HTML with "delayed" content blocks "popping in" via inline `<script>` tags later. [Read more](https://github.com/reactwg/react-18/discussions/37)
+Render a React element to its initial HTML without interaction. Returns a [Node.js stream](https://nodejs.dev/learn/nodejs-streams) with a `pipe(res)` method to pipe the output and `abort()` to abort the request.
 
-If you call [`ReactDOM.hydrateRoot()`](/docs/react-dom-client.html#hydrateroot) on a node that already has this server-rendered markup, React will preserve it and only attach event handlers, allowing you to have a very performant first-load experience.
+This method fully supports Suspense, and offers two approaches for handling suspended content. The first lets you stream content blocks as they become ready. The second waits until **all** suspended content blocks are ready before sending a response.
+
+#### Suspense approach one: Initial shell with streamed content blocks
+
+The first allows you to respond with an initial HTML shell and then "pop in" content blocks when they become ready via inline `<script>` tags later. [Read more on how this works](https://github.com/reactwg/react-18/discussions/37)
+
+To get this behaviour, implement the `onShellReady` callback and start streaming your response here. The `onShellError` option will be called if an error occurs before the shell was ready. The `onError` option will be called if an error occurs after the shell is ready.
 
 ```javascript
 let didError = false;
 const stream = renderToPipeableStream(
   <App />,
   {
+    onShellError(error) {
+      // Something errored before we could complete the shell so we emit an alternative shell.
+      // The stream can be safely ignored.
+      res.statusCode = 500;
+      res.send(
+        '<!doctype html><p>Loading...</p><script src="clientrender.js"></script>'
+      );
+    }
     onShellReady() {
       // The content above all Suspense boundaries is ready.
       // If something errored before we started streaming, we set the error code appropriately.
       res.statusCode = didError ? 500 : 200;
       res.setHeader('Content-type', 'text/html');
       stream.pipe(res);
     },
-    onShellError(error) {
-      // Something errored before we could complete the shell so we emit an alternative shell.
-      res.statusCode = 500;
-      res.send(
-        '<!doctype html><p>Loading...</p><script src="clientrender.js"></script>'
-      );
+    onError(err) {
+      // TODO: when is this called?
```

Comment:
I don’t quite understand when this is called — is it after the shell is ready? So if a suspended content blocks produce an error?